transport-native-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollStreamChannel.java

```diff
@@ -824,7 +824,7 @@ void epollInReady() {
                     allocHandle.incMessagesRead(1);
                     pipeline.fireChannelRead(byteBuf);
                     byteBuf = null;
-                } while (allocHandle.continueReading());
+                } while (config.isAutoRead() && allocHandle.continueReading());
```

Comment:
> The RecvByteBufAllocator.DelegatingHandle Epoll implementations will completely disregard the value of config.isAutoRead() when in EDGE mode

@blucas - Auto read is intentionally ignored because of the way EPOLL ET works. If ET is enabled and we don't read all available data, then we may never be notified again there is data pending (see [Level-triggered and edge-triggered](http://man7.org/linux/man-pages/man7/epoll.7.html)).